docs/builder.rst

```diff
@@ -0,0 +1,123 @@
+Builders
+========
+
+A STAC Item can be thought of as one or more Assets with additional metadata,
+some of which is derived from those assets. In many cases, creating an item from
+geospatial assets is highly repeatable. The :py:mod:`stactools.core.builder`
+module aims to abstract away many of the commonly-repeated operations required
+to create items from assets.
+
+Base class
+----------
+
+The base class, :py:class:`stactools.core.builder.Builder`, is pretty useless on
+its own. It takes a dictionary of assets as its initializer, and wraps those
+assets in an Item with a provided id and an auto-generated datetime::
+
+    from stactools.core.builder import Builder
+    from pystac import Asset
+
+    builder = Builder()
+    builder.add_asset("an-asset", Asset(href="a-file.dat"))
```

Comment:
Style question: what's the motivation for `builder = Builder(); builder.add_asset` pattern rather than defining what `Builder.create_item` will do in the class definitation?

Reply:
Partially, it's my procedural background, but I do find that leaving the asset addition until later is useful for conditionals, see here: https://github.com/stactools-packages/modis/blob/524bfd26bb6d571336e50a226e853b97287529e3/src/stactools/modis/commands.py#L74-85.